### PR TITLE
Api structure

### PR DESCRIPTION
**Pull request by JosueRodriguezGarcia**

### CODE REVIEW

File: src/test/resources/features/list.feature [hunk 1]
```diff
@@ -1,8 +1,8 @@
...
-    When I select TestBoard board
-    And I create a new list with "TestList" as title
-    Then I should see the new created list with the given title
\ No newline at end of file
+#  @ArchiveList @LogOut
+#  Scenario: New list creation
+#    Given I log in as admin user
+#    When I select TestBoard board
+#    And I create a new list with "TestList" as title
+#    Then I should see the new created list with the given title
```
Comment (MelissaRoman06) [inline src/test/resources/features/list.feature]: Please add an empty line at the end of the class.

File: src/main/java/trello/api/rest/Authentication.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+    /**
+     * Private constructor requested by checkstyle.
+     */
+    private Authentication() {
+
+    }
+
+    /**
+     * Gets the RequestSpecification for the authentication type oauth.
```
Comment (raulchoque-eng) [inline src/main/java/trello/api/rest/Authentication.java:39]: Please add a white space bellow this line.

File: src/main/java/trello/api/rest/Authentication.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+     * @return a RequestSpecification.
+     */
+    public static RequestSpecification getRequestSpecification() {
```
Comment (raulchoque-eng) [inline src/main/java/trello/api/rest/Authentication.java]: A think this method we can search a parameter with the type of user. It was the user that we get from json file.

File: src/main/java/trello/api/rest/RestClientAPI.java [hunk 4]
```diff
@@ -12,57 +12,106 @@
...
-import trello.keys.NamePages;
+
+import java.util.Map;
+
+import static io.restassured.RestAssured.given;
+import static io.restassured.http.ContentType.JSON;
 
 /**
- * RestClientAPI class.
+ * Authentication class.
```
Comment (raulchoque-eng) [inline src/main/java/trello/api/rest/RestClientAPI.java]: Please, change the name of this class we are in the RestClientAPI.

File: src/main/java/trello/api/rest/RestClientAPI.java [hunk 5]
```diff
@@ -12,57 +12,106 @@
...
  *
- * @author Josue Rodriguez.
- * @version 0.0.1
+ * @author Josue Rodriguez Garcia.
+ * @version 0.0.1.
  */
-public final class RestClientAPI {
+public class RestClientAPI {
+    private RequestSpecification requestSpecification;
```
Comment (raulchoque-eng) [inline src/main/java/trello/api/rest/RestClientAPI.java:32]: Please add a white space above this line.

File: src/main/java/trello/api/rest/RestClientAPI.java [hunk 6]
```diff
@@ -12,57 +12,106 @@
...
+
+    /**
+     * Constructor method.
```
Comment (raulchoque-eng) [inline src/main/java/trello/api/rest/RestClientAPI.java:35]: Please add a white space bellow this line.

File: src/main/java/trello/api/rest/RestClientAPI.java [hunk 7]
```diff
@@ -12,57 +12,106 @@
...
+     * @param requestSpecification defines a requestSpecification for input.
+     */
+    public RestClientAPI(final RequestSpecification requestSpecification) {
+        this.requestSpecification = requestSpecification;
+    }
+
+    /**
+     * Gets a response of type get.
```
Comment (raulchoque-eng) [inline src/main/java/trello/api/rest/RestClientAPI.java:44]: Please add a white space bellow this line.

File: src/main/java/trello/api/rest/RestClientAPI.java [hunk 8]
```diff
@@ -12,57 +12,106 @@
...
+     * @param endpoint defines the type the http method.
+     * @return a response.
+     */
+    public Response get(final String endpoint) {
+        return apiResponse("GET", endpoint);
+    }
+
+    /**
+     * Gets a response of type put.
```
Comment (raulchoque-eng) [inline src/main/java/trello/api/rest/RestClientAPI.java:54]: Please add a white space bellow this line.

File: src/main/java/trello/api/rest/RestClientAPI.java [hunk 9]
```diff
@@ -12,57 +12,106 @@
...
+     * @return a response.
+     */
+    public Response put(final String endpoint) {
+        return apiResponse("PUT", endpoint);
+    }
 
-    private static RestClientAPI oauth;
-    private RequestSpecification request;
+    /**
+     * Gets a response of type post.
```
Comment (raulchoque-eng) [inline src/main/java/trello/api/rest/RestClientAPI.java:64]: Please add a white space bellow this line.

File: src/main/java/trello/api/rest/RestClientAPI.java [hunk 10]
```diff
@@ -12,57 +12,106 @@
...
+     * @param endpoint defines the type the http method.
+     * @return a response.
+     */
+    public Response post(final String endpoint) {
+        return apiResponse("POST", endpoint);
+    }
 
     /**
-     * This is constructor that initializes variables.
+     * Gets a response of type delete.
```
Comment (raulchoque-eng) [inline src/main/java/trello/api/rest/RestClientAPI.java:74]: Please add a white space bellow this line.

File: src/main/java/trello/api/rest/RestClientAPI.java [hunk 11]
```diff
@@ -12,57 +12,106 @@
...
-                accessToken, tokenSecret)).setBaseUri(baseUrl).build();
+    public Response delete(final String endpoint) {
+        return apiResponse("DELETE", endpoint);
     }
 
     /**
-     * Gives the class instance according Singleton pattern.
-     *
-     * @return an instance.
+     * Gets a requestSpecification for RestClientAPI.
```
Comment (raulchoque-eng) [inline src/main/java/trello/api/rest/RestClientAPI.java:84]: Please add a white space bellow this line.

File: src/main/java/trello/api/rest/RestClientAPI.java [hunk 12]
```diff
@@ -12,57 +12,106 @@
...
-        return oauth;
+    public RequestSpecification getRequest() {
+        return requestSpecification;
     }
 
     /**
-     * Gives the request specification resultant of oauth.
-     *
-     * @return an request specification.
+     * Sets the attribute requestSpecification en RestClientAPI.
```
Comment (raulchoque-eng) [inline src/main/java/trello/api/rest/RestClientAPI.java:93]: Please add a white space bellow this line.

File: src/main/java/trello/api/rest/RestClientAPI.java [hunk 13]
```diff
@@ -12,57 +12,106 @@
...
+     * @param requestSpecification defines a requestSpecification in attribute to be set.
      */
-    public RequestSpecification getRequestSpecification() {
-        return request;
+    public void setRequest(final RequestSpecification requestSpecification) {
+        this.requestSpecification = requestSpecification;
+    }
+
+    /**
+     * Builds the request defines the method http and endpoint.
```
Comment (raulchoque-eng) [inline src/main/java/trello/api/rest/RestClientAPI.java:102]: Please add a white space bellow this line.

File: src/main/java/trello/api/rest/RestClientAPI.java [hunk 14]
```diff
@@ -12,57 +12,106 @@
...
+     */
+    private Response apiResponse(final String httpMethod, final String endpoint) {
+        return given().
+                spec(requestSpecification).
+                when().
+                request(httpMethod, endpoint);
+    }
+
+    /**
+     * Builds the request with a body.
```
Comment (raulchoque-eng) [inline src/main/java/trello/api/rest/RestClientAPI.java:116]: Please add a white space bellow this line.

File: src/test/java/trello/hooks/api/BoardAPIHook.java [hunk 15]
```diff
@@ -30,7 +32,10 @@
 
     private Context context;
     private Response response;
-    private RequestManager requestManager;
+    private RestClientAPI request;
+    private final int orderAfter = 3;
+    private final int orderBefore = 1;
+//    private RequestSpecification requestSpecification;
```
Comment (raulchoque-eng) [inline src/test/java/trello/hooks/api/BoardAPIHook.java]: Please, clear or decomment this line.

Comment (JosueRodriguezGarcia) [reply src/test/java/trello/hooks/api/BoardAPIHook.java]: Solved

File: src/main/java/trello/api/rest/RestClientAPI.java (hunk 14 above)
Comment (JosueRodriguezGarcia) [reply src/main/java/trello/api/rest/RestClientAPI.java:116]: Solved

File: src/main/java/trello/api/rest/RestClientAPI.java (hunk 13 above)
Comment (JosueRodriguezGarcia) [reply src/main/java/trello/api/rest/RestClientAPI.java:102]: Solved

File: src/main/java/trello/api/rest/RestClientAPI.java (hunk 12 above)
Comment (JosueRodriguezGarcia) [reply src/main/java/trello/api/rest/RestClientAPI.java:93]: Solved

File: src/main/java/trello/api/rest/RestClientAPI.java (hunk 11 above)
Comment (JosueRodriguezGarcia) [reply src/main/java/trello/api/rest/RestClientAPI.java:84]: Solved

File: src/main/java/trello/api/rest/RestClientAPI.java (hunk 10 above)
Comment (JosueRodriguezGarcia) [reply src/main/java/trello/api/rest/RestClientAPI.java:74]: Solved

File: src/main/java/trello/api/rest/RestClientAPI.java (hunk 9 above)
Comment (JosueRodriguezGarcia) [reply src/main/java/trello/api/rest/RestClientAPI.java:64]: Solved

File: src/main/java/trello/api/rest/RestClientAPI.java (hunk 8 above)
Comment (JosueRodriguezGarcia) [reply src/main/java/trello/api/rest/RestClientAPI.java:54]: Solved

File: src/main/java/trello/api/rest/RestClientAPI.java (hunk 7 above)
Comment (JosueRodriguezGarcia) [reply src/main/java/trello/api/rest/RestClientAPI.java:44]: Solved

File: src/main/java/trello/api/rest/RestClientAPI.java (hunk 6 above)
Comment (JosueRodriguezGarcia) [reply src/main/java/trello/api/rest/RestClientAPI.java:35]: Solved

File: src/main/java/trello/api/rest/RestClientAPI.java (hunk 5 above)
Comment (JosueRodriguezGarcia) [reply src/main/java/trello/api/rest/RestClientAPI.java:32]: Solved

File: src/main/java/trello/api/rest/RestClientAPI.java (hunk 4 above)
Comment (JosueRodriguezGarcia) [reply src/main/java/trello/api/rest/RestClientAPI.java]: Solved

File: src/main/java/trello/api/rest/Authentication.java (hunk 3 above)
Comment (JosueRodriguezGarcia) [reply src/main/java/trello/api/rest/Authentication.java]: Solved

File: src/main/java/trello/api/rest/Authentication.java (hunk 2 above)
Comment (JosueRodriguezGarcia) [reply src/main/java/trello/api/rest/Authentication.java:39]: Solved

File: src/test/resources/features/list.feature (hunk 1 above)
Comment (JosueRodriguezGarcia) [reply src/test/resources/features/list.feature]: Solved

**Review by raulchoque-eng — APPROVED** [review]

**Pull request merged**